### PR TITLE
Add OSX job to azure

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,31 +1,74 @@
...
+  - checkout: self
+    fetchDepth: 10
+
+  - script: |
+      set -e
+      brew update
+      brew unlink python
+      brew install gnu-time opam
+
+      opam init -y -a -j 2 --compiler=$COMPILER
```
Comment (ejgallego) [inline azure-pipelines.yml]: All the opam commands should use the OPAM 2 format.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,31 +1,74 @@
...
+    env:
+      COMPILER: "4.07.0"
+      FINDLIB_VER: ".1.8.0"
+
+  - script: |
+      set -e
+
+      eval $(opam config env)
+      ./configure -local -warn-error yes -native-compiler no
+      make -j 2
```
Comment (ejgallego) [inline azure-pipelines.yml]: Number of cores should not be hardcoded but the value of `NJBOS` [or the corresponding Azure var if available] should be used.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -1,31 +1,74 @@
...
+    displayName: 'Build Coq'
+
+  - script: |
+      eval $(opam config env)
+      make -j 2 test-suite
+    displayName: 'Test Coq'
```
Comment (ejgallego) [inline azure-pipelines.yml]: Maybe this should read more specific? "Coq Test Suite" ["Test" seems too generic to me]

File: azure-pipelines.yml (hunk 1 above)
Comment (SkySkimmer) [reply azure-pipelines.yml]: >All the opam commands should use the OPAM 2 format.

I don't know what that means.

Comment (ejgallego) [reply azure-pipelines.yml]: `Dockerfile` contains OPAM 2 format commands.

Comment (ejgallego) [inline azure-pipelines.yml]: add `--disable-sandboxing`

File: azure-pipelines.yml [hunk 4]
```diff
@@ -1,31 +1,74 @@
...
+    fetchDepth: 10
+
+  - script: |
+      set -e
+      brew update
+      brew unlink python
+      brew install gnu-time opam
+
+      opam init -y -a -j 2 --compiler=$COMPILER
+      opam switch $COMPILER
```
Comment (ejgallego) [inline azure-pipelines.yml]: Should be `opam switch set` [or `create`.]

File: azure-pipelines.yml [hunk 5]
```diff
@@ -1,31 +1,74 @@
...
+      eval $(opam config env)
```
Comment (ejgallego) [inline azure-pipelines.yml]: Should be `eval $(opam env)`

File: azure-pipelines.yml [hunk 6]
```diff
@@ -1,31 +1,74 @@
...
+      opam update
+      opam install -j 2 -y num ocamlfind${FINDLIB_VER} ounit
```
Comment (ejgallego) [inline azure-pipelines.yml]: Remove `-y` and export `OPAMYES: true` instead.

File: azure-pipelines.yml [hunk 7]
```diff
@@ -1,31 +1,74 @@
...
+      opam list
+    displayName: 'Install dependencies'
+    env:
+      COMPILER: "4.07.0"
+      FINDLIB_VER: ".1.8.0"
+
+  - script: |
+      set -e
+
+      eval $(opam config env)
```
Comment (ejgallego) [inline azure-pipelines.yml]: Ditto.

File: azure-pipelines.yml [hunk 8]
```diff
@@ -1,31 +1,74 @@
...
+      ./configure -local -warn-error yes -native-compiler no
+      make -j 2
+    displayName: 'Build Coq'
+
+  - script: |
+      eval $(opam config env)
```
Comment (ejgallego) [inline azure-pipelines.yml]: Ditto.

**Comment by SkySkimmer** [comment]
@Zimmi48 https://gitlab.com/coq/coq/-/jobs/143593562 looks like a spurious timeout, should the bot have caught it? (I manually restarted)

File: azure-pipelines.yml (hunk 1 above)
Comment (SkySkimmer) [reply azure-pipelines.yml]: >add --disable-sandboxing

Why? It works without.

Comment (ejgallego) [reply azure-pipelines.yml]: I understand that makes the build faster, also it reduces the complexity of the dependency chain, but we can leave it there if you want.

**Comment by SkySkimmer** [comment]
Changed

**Comment by Zimmi48** [comment]
>https://gitlab.com/coq/coq/-/jobs/143593562 looks like a spurious timeout, should the bot have caught it? (I manually restarted)

Hum, since when are timeouts spurious? BTW, it timed out again!
(I mean, I don't think the bot should restart this test in the same way that it doesn't restart failed tests due to non-deterministic tests. We should fix these instead.)

**Review by ejgallego — APPROVED** [review]
Looks good, thanks, and sorry for the delay.

**Pull request merged**